spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -310,18 +310,21 @@ private IntegrationResourceHolder bindResourceHolderIfNecessary(String key, Obje
 		if (this.transactionSynchronizationFactory != null && resource != null &&
 				TransactionSynchronizationManager.isActualTransactionActive()) {
 			TransactionSynchronization synchronization = this.transactionSynchronizationFactory.create(resource);
-			TransactionSynchronizationManager.registerSynchronization(synchronization);
```

Comment:
With your suggestion we change behavior when the `TransactionSynchronization` is not `null` and not `IntegrationResourceHolderSynchronization`. Right now we register it independently of its type.